File: user/src/main/scala/ru/tdd/app/models/dto/ExceptionDto.scala
package ru.tdd.app.models.dto

import ru.tdd.app.models.exceptions.ApiException
import zio.json.{DeriveJsonDecoder, DeriveJsonEncoder, JsonDecoder, JsonEncoder}

import java.time.LocalDateTime

/**
 * @author Tribushko Danil
 * @since 05.11.2025
 * @param message   текст ошибки
 * @param timestamp время ошибки
 *                  Dto-объект исключения
 */
case class ExceptionDto(
                         message: String,
                         timestamp: LocalDateTime
                       )

object ExceptionDto {

  implicit val encoder: JsonEncoder[ExceptionDto] = DeriveJsonEncoder.gen[ExceptionDto]
  implicit val decoder: JsonDecoder[ExceptionDto] = DeriveJsonDecoder.gen[ExceptionDto]

  def apply(ex: ApiException): ExceptionDto =
    ExceptionDto(ex.getMessage, ex.timestamp)
}

File: geo/src/main/java/ru/tdd/geo/application/models/dto/ExceptionDTO.java
package ru.tdd.geo.application.models.dto;

/**
 * @author Tribushko Danil
 * @since 07.01.2026
 * DTO исключения приложения
 */
public class ExceptionDTO {

    private int statusCode;

    private String message;

    public ExceptionDTO() {}

    public ExceptionDTO(int statusCode, String message) {
        this.statusCode = statusCode;
        this.message = message;
    }

    public int getStatusCode() {
        return statusCode;
    }

    public void setStatusCode(int statusCode) {
        this.statusCode = statusCode;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }
}
